src/Stratis.Bitcoin.Features.BlockStore/BlockStoreQueue.cs

```diff
@@ -86,6 +90,7 @@ public class BlockStoreQueue : IBlockStore, IDisposable
             this.storeSettings = storeSettings;
             this.chain = chain;
             this.blockRepository = blockRepository;
+            this.batch = new List<ChainedHeaderBlock>();
```

Comment:
Where do you add/remove to the list?

Reply:
On `DequeueBlocksContinuouslyAsync`

I just moved batch list creation from the method body to a class variable to get it's size in the logs.